### ISSUE TITLE
some infos please...

### ISSUE THREAD

**Issue by 7starsone** [TwinFuture/virtualmin-nginx#1]
could you add a detailed readme please...?
what are possible issues at the moment and how to manage the existing vhosts previously on apache?

thanks

**Comment by TwinFuture** [comment]
Hey 7starsone, there are currently no issues that I have found in the code from what I have experienced.
If you do find any please let me know!

There are some additions I have planned for the code to manage the php-fpm aspect within a control panel.

Managing and creating is pretty much the same as the previous version apart from some code/bug fixes and the addition to creating a php-fpm pool file for the server created. I modified this plugin to support php-fpm auto creation. Managing the php-fpm part for each virtual host has to be done manually. 

I will try and provide some more documentation on changes when I get some time. I feel I covered most of what is fixed within the post made on Virtualmin. It could be expanded.

Thanks for the feedback. 
If there is anything you are unsure of please let us know.

**Comment by 7starsone** [comment]
@TwinFuture  "Managing the php-fpm part for each virtual host has to be done manually" this is a big problem, above all because you need to consider that a server admin may have many domains to manage...that's impossible to do manually. Thank you

**Comment by TwinFuture** [comment]
Yeah I understand that.. Manually means you would have to open up the pool.conf file and append changes manually to your virtual host. Originally I just created this to fix some issues with backing up, renaming and other bugs with the default version and to create Nginx and PHP-FPM config files automatically and your virtual host will be instantly usable by PHP. Rather than having to create/edit these files manually to setup a host to run php-fpm and nginx. 

 I do plan on adding a GUI panel for PHP-FPM to configure zones and other configuration settings for php-fpm like you can currently with Nginx. A PHP-FPM panel to modify the settings for your virtual host would be a nice feature, but I don't see how it would save time as you still have to view each hosts file, GUI or text, and append the changes to each host. A GUI Panel could make things easier and faster to manage I agree. I guess we can request this as a feature. 

Thanks again.

**Comment by kintaro1981** [comment]
notice:
virtualmin now include php-fpm official support
it has only some little bug that are going to be fixed in the next release
https://www.virtualmin.com/project/issues?text=php-fpm&projects=&status=Open&categories=All&order=created&sort=desc

**Comment by TwinFuture** [comment]
@kintaro1981 I did download their latest build and couldn't find where to enable php-fpm. Maybe I am missing something? 2.1

I think its to do with the php56-php-fpm I have installed its not noticing it. I will stick with what I have for now as I can't find where to enable this as the options in PHP script execution mode only show a FCGId (run as virtual server owner) option.

Also did they fix the bugs I found? Things like backing up a virtual server nginx configuration file when it contains multiple server blocks? Renaming the virtual server and many other things?

**Comment by kintaro1981** [comment]
Server Configuration => Website Options

But actually it end up with a permission error, there is already an issue open on their bug tracker about that.
